File: app/src/main/java/ua/te/hackathon/smartcity2015/ui/base/adapters/BaseRecyclerAdapter.java
package ua.te.hackathon.smartcity2015.ui.base.adapters;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;

import java.util.List;

/**
 * @author victor
 * @since 2016-02-14
 */
public abstract class BaseRecyclerAdapter<T, V extends RecyclerView.ViewHolder> extends RecyclerView.Adapter<V> {

  private List<T> itemList = null;

  public void setItemList(@NonNull List<T> itemList) {
    this.itemList = itemList;
  }

  public T getItem(int pos) {
    return itemList == null ? null : itemList.get(pos);
  }

  @Override
  public int getItemCount() {
    return itemList == null ? 0 : itemList.size();
  }
}

File: app/src/main/java/ua/te/hackathon/smartcity2015/ui/base/adapters/BaseRecyclerAdapter.kt
package ua.te.hackathon.smartcity2015.ui.base.adapters

import android.support.v7.widget.RecyclerView

/**
 * @author victor
 * *
 * @since 2016-02-14
 */
abstract class BaseRecyclerAdapter<T, V : RecyclerView.ViewHolder> : RecyclerView.Adapter<V>() {

  private var itemList: List<T>? = null

  fun setItemList(itemList: List<T>) {
    this.itemList = itemList
  }

  fun getItem(pos: Int): T? {
    return if (itemList == null) null else itemList!![pos]
  }

  override fun getItemCount(): Int {
    return if (itemList == null) 0 else itemList!!.size
  }
}
